iothub/device/src/DeviceClient.cs

```diff
@@ -571,27 +561,6 @@ public void SetRetryPolicy(IRetryPolicy retryPolicy)
         public Task SendEventBatchAsync(IEnumerable<Message> messages, CancellationToken cancellationToken) =>
             InternalClient.SendEventBatchAsync(messages, cancellationToken);
 
-        /// <summary>
-        /// Uploads a stream to a block blob in a storage account associated with the IoTHub for that device.
-        /// If the blob already exists, it will be overwritten.
-        /// </summary>
-        /// <param name="blobName">The name of the blob to upload.</param>
-        /// <param name="source">A stream with blob contents. Should be disposed after upload completes.</param>
-        [Obsolete("This API has been split into three APIs: GetFileUploadSasUri, uploading to blob directly using the Azure Storage SDK, and CompleteFileUploadAsync")]
-        public Task UploadToBlobAsync(string blobName, Stream source) => InternalClient.UploadToBlobAsync(blobName, source);
```

Comment:
Another improvement would be to consolidate the cancellation token and non cancellation token variants of the methods into a single method with an optional cancellation token. You could take it up in a different PR though.